docs/source/tutorials/profiling.ipynb

```diff
@@ -5,11 +5,375 @@
+    "\n",
+    "Understanding all the above requires a careful consideration of RP's architecture and an understanding of where and when each component executes. Sounds too complicated? It is. Go and see how to use [RADICAL-Analytics](https://radicalanalytics.readthedocs.io/en/latest/) to simplify tracing and profiling, albeit only after your application has completed its execution.\n",
+    "\n",
+    "In RP's lingo, a state is a special type of event. States have two properties that are relevant to tracing:\n",
+    "\n",
+    "1. Each state is guaranteed to fire in every possible configuration of RP, i.e., independent on the platform on which it executes, the number and types of resources, the specific virtual environment it uses for its Agent and the other myriad of possible configurations of RP.\n",
+    "2. States are guaranteed to always fire in a precise sequence, i.e., each state is part of a well-defined, immutable sequence.\n",
+    "\n",
+    "When tracing your application, states trade off granularity for reliability. As such, they are also easier to use as they require a less granular and specific understanding of RP. To see the list of states, ordered in their guaranteed sequence, see RP's [state model](../internals.rst#state-model).\n",
+    "\n",
+    "## Record format\n",
+    "\n",
+    "Each `*.prof` file uses the following format:\n",
+    "\n",
+    "```\n",
+    "time,event,comp,thread,uid,state,msg\n",
+    "```\n",
+    "\n",
+    "Where:\n",
+    "\n",
+    "- `time` = (mandatory) timestamp expressed in Unix Time (epoch)\n",
+    "- `event` = (mandatory) the name of the timestamped event\n",
+    "- `comp` = (facultative) the name of the component that triggered the event\n",
+    "- `thread` = (mandatory) the ID of the thread in which the event was triggered with `comp`\n",
+    "- `uid` = (mandatory) the ID of the entity to which that event belongs (e.g., task or pilot)\n",
+    "- `state` = (facultative) the name given to the event when it is also a state, i.e., guaranteed to be triggered in every configuration of RP\n",
+    "- `msg` = (facultative) an arbitrary message associated to that event.\n",
+    "\n",
+    "For example, here is the record of the `launch_start` event for a task:\n",
+    "\n",
+    "```\n",
+    "1684787127.2228520,launch_start,,MainThread,task.000000,AGENT_EXECUTING,\n",
+    "```\n",
+    "\n",
+    "As you will see later in this tutorial, you can use `launch_start` to monitor when a task is/was executed. Here is another example, this time for the `schedule_ok` event triggered by RP's Agent Scheduler:\n",
```

Comment:
`to monitor when a task is/was executed` -> `to monitor when a task is/was launched to be executed`?